core/web/auth/gql.go

```diff
@@ -10,6 +10,10 @@ import (
 )
 
 type sessionUserKey struct{}
+type GQLQUserSession struct {
```

Comment:
We don't seem to need this anymore given we are using  `auth. GetGQLAuthenticatedUser`

Reply:
I created this struct to be able to return both the user + the session ID, so at the end of the day, we are storing this struct in the context and consuming it when the user tries to change its password.

Since we need to remove all sessions but this one, I needed this struct to access the value of the sessionID and pass it down to `ClearNonCurrentSessions`.

Reply:
OK I see. Then we should probably do a bit of refactoring to make this clearer.

For example it is still using the name Set/GetAutheticatedUser. Rename and change arg signature to reflect we are storing a session.

`SetGQLAuthenticatedSession(ctx context.Context, GQLSession)`

`GetGqlAuthenticatedSession(ctx context.Context)`